source/common/http/conn_manager_impl.h

```diff
@@ -192,7 +192,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void maybeEndDecode(bool end_stream);
 
     // Http::RequestDecoder
-    void decodeHeaders(RequestHeaderMapPtr&& headers, bool end_stream) override;
+    void decodeHeaders(RequestHeaderMapSharedPtr&& headers, bool end_stream) override;
```

Comment:
Is this interface change needed? It can take a unique_ptr and store the pointee in a shared_ptr.

Reply:
Agreed that keeping the interface unchanged is preferable.

I tried changing the interface back to accept a unique ptr, and leaving the member variable as a shared ptr, but ran into the problem that in ActiveStream::RecreateStream we call decodeHeaders on the request_headers_ member variable: https://github.com/envoyproxy/envoy/blob/3752119b94ae4a5ff337364f585f54305406dc7f/source/common/http/conn_manager_impl.cc#L1788. I'm not sure if there is a simple solution to this that I'm not seeing, do you have any suggestion?